pkg/combustion/rpm.go

```diff
@@ -110,3 +104,48 @@ func writeRPMScript(ctx *image.Context, rpmFileNames []string) (string, error) {
 
 	return modifyRPMScriptName, nil
 }
+
+// determine whether RPM configuration is needed
+func skipRPMconfigre(ctx *image.Context) bool {
+	pkg := ctx.ImageDefinition.OperatingSystem.Packages
+
+	if isComponentConfigured(ctx, userRPMsDir) && len(pkg.AddRepos) > 0 {
+		// is RPM configured in 'rpms' directory from a third party repository
+		return false
+	} else if isComponentConfigured(ctx, userRPMsDir) {
+		// is RPM configured as a standalone RPM inside of the 'rpms' directory
+		return false
+	}
+
+	// is package configured for installation from PackageHub
+	if len(pkg.PKGList) > 0 && pkg.RegCode != "" {
+		return false
+	}
+
+	// is package configured for installation from a third party repository
+	if len(pkg.AddRepos) > 0 && len(pkg.PKGList) > 0 {
+		return false
+	}
```

Comment:
I appreciate that we have a whole bunch of different scenarios but I don't believe code duplication is the way to distinguish between those. This function only wants to make the call if the component should be skipped or not.
```suggestion
        if isComponentConfigured(ctx, userRPMsDir) {
		// User provided standalone or third party RPMs
		return false
	}

	if len(pkg.PKGList) > 0 {
		if pkg.RegCode != "" {
			// Packages requested from PackageHub
			return false
		}

		if len(pkg.AddRepos) > 0 {
			// Packages requested from third party repositories
			return false
		}
	}
```

Reply:
I agree, this has further been simplified since the addition of the EBI image definition `validation` logic.